### ISSUE TITLE
Preserve comments in code?

### ISSUE THREAD

**Issue by yanickrochon** [bholloway/esmangleify#1]
Is it possible to prevent comments from being dropped?

**Comment by bholloway** [comment]
I don't think escodegen supports it but I'll check.

Are you mainly concerned about the header? I can potentially patch that in.

Let me know of you are working on a pull request.

**Comment by yanickrochon** [comment]
My concern was mostly to perform some token replacement. Basically, I browserify some shared modules (client/server) then bundle these statically and cache the result. Then, when the client HTTP GET a certain URL, I just return that cached result as a `text/javascript` content, replacing some tokens with dynamic data, specific to the requiring client. And since I was using comments like `/*{{token}}*/`  to do that, and they were removed, it broke my initial design.

Following that, I started using the [replace](https://github.com/rse/browserify-replace) transform to replace these tokens before going through esmangleify with some external data object references, which I inserted using `JSON.stringify` before the cached result.

Perhaps some ppl may be interested into keeping the header comments... I think I'll leave my implementation as it is. :)
